include/triton/Dialect/Triton/IR/TritonTypes.td

```diff
@@ -102,6 +102,28 @@ def TT_TensorDescType : TritonTypeDef<"TensorDesc", "tensordesc", []> {
 
   let parameters = (ins "RankedTensorType":$blockType);
   let assemblyFormat = "`<` $blockType `>`";
+
+  let builders = [
+    TypeBuilder<(ins "RankedTensorType":$blockType, "bool":$isSigned), [{
+      if (isSigned) {
+        auto intTy = llvm::cast<IntegerType>(blockType.getElementType());
```

Comment:
This is slightly awkward, since this implies we treat the "signless" `i32` as unsigned, and `si32` as signed. WDYT about keeping an `isSigned` flag on the descriptor type instead? I think that would help keep some of APIs cleaner as well.

Reply:
I'm not a fan since it means we need the extra `isSigned` in the IR format, whereas this way it just shows up in the type and only for ints. For consistency, I've changed it to use `ui` for unsigned types though.

Reply:
works for me